internal/olm/operator/bundle/install.go

```diff
@@ -98,3 +185,137 @@ func (i *Install) setup(ctx context.Context) error {
 
 	return nil
 }
+
+// addBundleToIndexImage adds the bundle to an existing index image if the bundle is not already present in the index image.
+func addBundleToIndexImage(ctx context.Context, indexImage string, bundleDeclConfig *declarativeconfig.DeclarativeConfig) (*declarativeconfig.DeclarativeConfig, error) {
+	log.Infof("Rendering a File-Based Catalog of the Index Image %q", indexImage)
+	log.SetOutput(ioutil.Discard)
+	render := action.Render{
+		Refs: []string{indexImage},
+	}
+
+	imageDeclConfig, err := render.Run(ctx)
+	if err != nil {
+		log.Errorf("Error in rendering the index image %q: %v", indexImage, err)
+		return nil, err
+	}
+	log.SetOutput(os.Stdout)
+
+	if len(bundleDeclConfig.Bundles) < 0 {
+		return nil, fmt.Errorf("bundles rendered are less than 0 for the bundle image")
+	}
+
+	// check if the bundle exists in the index image.
+	if len(bundleDeclConfig.Channels) > 0 && len(bundleDeclConfig.Bundles) > 0 {
+		for _, channel := range imageDeclConfig.Channels {
+			// find the specific channel that the bundle needs to be inserted into.
+			if channel.Name == bundleDeclConfig.Channels[0].Name && channel.Package == bundleDeclConfig.Channels[0].Package {
+				// check if the CSV name is already present in the channel's entries.
+				for _, entry := range channel.Entries {
+					if entry.Name == bundleDeclConfig.Bundles[0].Name {
+						return nil, fmt.Errorf("bundle %q already present in the index image: %s", bundleDeclConfig.Bundles[0].Name, indexImage)
+					}
+				}
+
+				break // we only want to search through the specific channel.
+			}
+		}
+	}
```

Comment:
Can this be simplified to just check `olm.bundle` blobs in the index image? I feel like if the index image contains the bundle _anywhere_ we should fail out, but its possible I'm missing some context for why we're only checking in the specific channel.

Reply:
From my understanding, we should fail out only when the bundle that we are trying to install already exists in the index image, which is consistent with our existing implementation with the sqlite index images where the deprecated `opm registry add` command fails if the bundle we intend to install is present on the index. An index image can have the `olm.bundle` blob with for ex: `memcached-operator.v0.0.1` but we should still be able to install `memcached-operator.v0.0.2` with `run bundle` right?

The reason we are iterating channels is that there can be multiple bundle versions under the same package name so instead we need to find whether the specific version of the bundle exists on the image. So for a matching channel in both bundle image declcfg and index image declcfg, we look for the specific csv name and if it exists, then we error out. Isn't this necessary to do?

Reply:
@joelanford I think the reason we didn't do something like

```
for _, bundle := range imageDeclConfig.Bundles {
   if bundle.Name == bundleDeclConfig.Bundles[0].Name && bundle.Package == bundleDeclConfig.Bundles[0].Package {
	// return error
   }
}
```

is because of the case where the bundle may exist, but in a different channel. Example: The index image has the memcached bundle in the "alpha" channel, but we want to the bundle to also be in the "stable" channel (which we are yet to create). The above snippet wouldn't catch that. 

@rashmigottipati for the above example, we'd only need to add a new channel (with this entry) and NOT append the package and bundle blobs, since those already exist (as @joelanford mentioned in [this](https://github.com/operator-framework/operator-sdk/pull/5671#discussion_r853416273) comment). I'm leaning towards having **individual checks** to see if packages, channels, and bundles exist in an index image, and appending accordingly. Curious as to what you think of this approach.

Reply:
Why operator-sdk run bundle allow us to inform an index?
What is the use? I mean, I understand that the operator-sdk run bundle will install the bundle and let us check the Operator installed via the bundle and using OLM.

So, what value do we bring by allowing a user to provide an index?
What are the use case scenarios? What SDK run bundle can do differ according to the index informed?

Shows me that the only extra test/check that we do here when the index is informed is to verify it would be possible to add the new bundle to a pre-existent index. Am I right? If yes, should it really be part of the run bundle scope and checks? Should not it only be used/useful for the upgrade bundle command?

I added it in : https://github.com/operator-framework/enhancements/pull/112/files#r856161326 as well

Reply:
@joelanford @VenkatRamaraju Modified the validation logic to insert the `bundle` into `index` only when `package`, `channel` and `bundle` blobs don't exist in the index. There's a case where a `bundle` already exists in a different channel and we would now like to add it to another channel, in which case, both bundle and package blobs will exist on the FBC and we need to add the bundle's channel to the index. 

Fixed in [commit](https://github.com/operator-framework/operator-sdk/pull/5671/commits/99f9c4c6c11d63cc78c1c2a3064227b489fa83c5).